app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java

```diff
@@ -0,0 +1,147 @@
+package openfoodfacts.github.scrachx.openfood.views;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.Toolbar;
+
+import com.mikepenz.materialize.color.Material;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import butterknife.BindView;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import openfoodfacts.github.scrachx.openfood.FastScroller;
+import openfoodfacts.github.scrachx.openfood.R;
+import openfoodfacts.github.scrachx.openfood.models.Additives;
+import openfoodfacts.github.scrachx.openfood.utils.SearchType;
+import openfoodfacts.github.scrachx.openfood.views.adapters.AdditivesAdapter;
+import openfoodfacts.github.scrachx.openfood.views.listeners.RecyclerItemClickListener;
+
+public class AdditivesExplorer extends BaseActivity implements AdditivesAdapter.ClickListener {
+
+
+    RecyclerView recyclerView;
+    ProgressBar progressBar;
+    FastScroller fastScroller;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_additives_explorer);
+
+        recyclerView = findViewById(R.id.additiveRecyclerView);
+        progressBar = findViewById(R.id.additives_progress_bar);
+        fastScroller = findViewById(R.id.additives_fast_scroller);
+
+
+        progressBar.setVisibility(View.VISIBLE);
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url("https://world.openfoodfacts.org/additives.json")
+                .build();
+
```

Comment:
@PrajwalM2212 there is no need for downloading data everytime as the additivies are already stored in local database in `AdditiveNmaeDao` and you can extract data from that using the `getAdditiveByTagAndLanguageCode` function in `ProductRepository.java` this even takes care of translations. By this you code will just become very compact.